lib/main.js

```diff
@@ -1,156 +1,98 @@
+    this.projectPath = projectPath;
 
-          try {
-            const results = await workerHelper.requestJob('fix', textEditor);
+    return {
+      processId: process.pid,
+      capabilities: {
+        workspace: {
+          configuration: true,
+        },
+      },
+      rootPath: projectPath,
+    };
+  }
 
-            const notificationText = results && results.length === 0 ?
-              'Linter-TSLint: Fix complete.' :
-              'Linter-TSLint: Fix attempt complete, but linting errors remain.';
+  startServerProcess(projectPath) {
+    this.projectPath = projectPath;
 
-            atom.notifications.addSuccess(notificationText);
-          } catch (err) {
-            atom.notifications.addWarning(err.message);
-          } finally {
-            // Restore cursor to the position before fix job
-            textEditor.setCursorBufferPosition(cursorPosition);
-          }
-        },
-      }),
+    return super.spawnChildNode(
+      [require.resolve('@keplersj/tslint-server/out/server')],
+      { cwd: projectPath, stdio: [null, null, null, 'ipc'] } // eslint-disable-line comma-dangle
     );
-
-    const createWorkerCallback = window.requestIdleCallback(() => {
-      this.worker = new Task(require.resolve('./worker.js'));
-      idleCallbacks.delete(createWorkerCallback);
+  }
+
+  preInitialization(connection) {
+    // connection._onNotification({ method: 'tslint/status' }, (params) => {
+    //   // This notification is used to set the status indicator in VSCode.
+    //
+    //   const status = params.state;
+    //   switch (status) {
+    //     case 1: // ok
+    //       atom.notifications.addSuccess("TSLint is OK");
+    //       break;
+    //     case 2: // warning
+    //       atom.notifications.addWarning("TSLint is not OK");
+    //       break;
+    //     case 3: // error
+    //       atom.notifications.addError("TSLint is not OK");
+    //       break;
+    //     default:
+    //       console.error("TSLint status changed in an unknown way!")
+    //   }
+    // });
+
+    connection._onRequest({ method: 'tslint/noLibrary' }, () => {
+      atom.notifications.addError("TSLint can't find TSLint Library in Project.");
     });
-    idleCallbacks.add(createWorkerCallback);
-  },
-
-  deactivate() {
-    idleCallbacks.forEach(callbackID => window.cancelIdleCallback(callbackID));
-    idleCallbacks.clear();
-    this.subscriptions.dispose();
-
-    workerHelper.terminateWorker();
-    this.worker = null;
-  },
-
-  provideLinter() {
-    return {
-      name: 'TSLint',
-      grammarScopes,
-      scope: 'file',
-      lintOnFly: true,
-      lint: async (textEditor) => {
-        if (this.ignoreTypings && textEditor.getPath().toLowerCase().endsWith('.d.ts')) {
-          return [];
-        }
 
-        if (!this.worker) {
-          await waitOnIdle();
-        }
+    const getConfig = () => ({
+      enable: true,
+      jsEnable: atom.config.get('linter-tslint.javascriptSupport'),
```

Comment:
You shouldn't have _any_ calls to `atom.config.get`, the `atom.config.observe` you have below registers a callback that is called when first initialized and then any time the setting is changed. In that callback you should be updating a local cache of the setting values. In this function here you should be simply returning the current state of that cache.